src/lib/languageUtils.ts

```diff
@@ -1,2 +1,77 @@
-export const CONFIG_NAME_REGEX = /[a-zA-Z0-9_ ]+/;
-export const CONFIG_NAME_WITH_QUOTES_REGEX = /["'`][a-zA-Z0-9_ ]+["'`]/;
+export const CONFIG_NAME_REGEX = /[a-zA-Z0-9_ ]+/g;
+export const CONFIG_NAME_WITH_QUOTES_REGEX = /["'`][a-zA-Z0-9_ ]+["'`]/g;
+
+const SUPPORTED_LANGUAGES = ['javascript', 'typescript'] as const;
+export type SupportedLanguageType = typeof SUPPORTED_LANGUAGES[number];
+export const SUPPORTED_FILE_EXTENSIONS = ['js', 'ts'];
+
+export function isLanguageSupported(language: string): boolean {
+  return (SUPPORTED_LANGUAGES as unknown as string[]).includes(language);
+}
+
+export function getStatsigAPICheckGateRegex(
+  language: SupportedLanguageType,
+  configName: string,
+): RegExp {
+  switch (language) {
+    case 'javascript':
+    case 'typescript':
```

Comment:
Would it be easier for you to have a `supportedLanguage` interface with all the functions in this file?

Reply:
Yep definitely. Will leave this restructuring to when I get around to supporting multiple languages.